### ISSUE TITLE
`--fix` ignores `yamllint disable` comments

### ISSUE THREAD

**Issue by mueller-ma** [ansible/ansible-lint#4250]
##### Summary

Sometimes it's easier to ignore the line length of one line than splitting it in multiple lines. For that yamllint comments can be used:

````yaml
# yamllint disable rule:line-length
foo: veeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeery-loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong-value-here
# yamllint enable rule:line-length
````

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 24.6.1 using ansible-core:2.17.1 ansible-compat:24.5.1 ruamel-yaml:0.18.5 ruamel-yaml-clib:0.2.8
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Run `ansible-lint --fix` on the yaml above.

##### Desired Behavior

Ansible-lint tries to "fix" this line and causes other issues by doing that.

##### Actual Behavior

Ansible-lint --fix should ignore that line as it doesn't cause an error.

**Comment by tanwigeetika1618** [comment]
Hi @mueller-ma , After reviewing your case and discussing it with the team, we have identified that the behavior you're experiencing—where ansible-lint --fix is not respecting the yamllint disable rule:line-length comments—is actually caused by the underlying YAML processing library, ruamel.yaml. This library is responsible for handling YAML formatting in the Ansible ecosystem, and it seems that the issue lies within how ruamel.yaml applies linting rules during the formatting process.
Since this issue is deeply tied to the internal workings of the ruamel.yaml library, it would be more effective to raise this concern directly with the maintainers of ruamel.yaml. When doing so, it would be helpful to mention any specific problems caused by the line-length changes, as this will aid in resolving the issue. They will be in the best position to address why the line-length disabling comment is being ignored and how it might be fixed in future updates. 
Therefore, I will proceed to close this issue based on the current findings.
